### PR TITLE
Remove non public functions from ContractDef

### LINKED ISSUE #19: Module's private function should not be consider as an entrypoint
Inside an implementation block, only public functions should be converted to entry points.

Currently, a module with a private function does not compile to a wasm.

### PR DESCRIPTION
**Pull request by kpob**
Fixes #19

### CODE REVIEW

**Review by kubaplas — APPROVED** [review]

File: lang/ir/src/module_item/method.rs [hunk 1]
```diff
@@ -9,6 +9,13 @@ pub struct Method {
     pub args: syn::punctuated::Punctuated<syn::PatType, syn::token::Comma>,
     pub ret: syn::ReturnType,
     pub full_sig: syn::Signature,
+    visibility: syn::Visibility,
```
Comment (zie1ony) [inline lang/ir/src/module_item/method.rs:12]:
```suggestion
    pub visibility: syn::Visibility,
```

File: lang/ir/src/module_item/module_impl.rs [hunk 2]
```diff
@@ -47,3 +58,28 @@ impl TryFrom<syn::ItemImpl> for ModuleImpl {
...
+
+                pub fn public_fn() {}
+
+                fn private_fn() {}
+            }
+        };
+        let module_impl = ModuleImpl::try_from(item_impl).unwrap();
+
+        assert_eq!(module_impl.methods().len(), 4);
+        assert_eq!(module_impl.public_methods().len(), 2);
```
Comment (zie1ony) [inline lang/ir/src/module_item/module_impl.rs:83]: `pub(crate)` is treated as private here, right?

**Review by zie1ony — APPROVED** [review]
+1

File: lang/ir/src/module_item/module_impl.rs (hunk 2 above)
Comment (kpob) [reply lang/ir/src/module_item/module_impl.rs:83]: Yup, only pure `pub`s are public.

**Pull request merged**